flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/EventSerializer.java

```diff
@@ -289,7 +289,7 @@ public static BufferConsumer toBufferConsumer(AbstractEvent event) throws IOExce
 
 		MemorySegment data = MemorySegmentFactory.wrap(serializedEvent.array());
 
-		return new BufferConsumer(data, FreeingBufferRecycler.INSTANCE, false);
+		return new BufferConsumer(data, FreeingBufferRecycler.INSTANCE, false, true);
```

Comment:
It is not always `true` for different cases calling `toBufferConsumer`. E.g. `ResultSubpartition#finish()` to generate `EndOfPartitionEvent`.
So we can have two options:

-  Provide the respective `isSharable` argument in the method of `toBufferConsumer`.  Then this tag in `BufferConsumer` is suitable for both buffer and event.

- The `isSharable` field in `BufferConsumer` is only indicating for buffer without event, then we do not need to touch this method to use default constructor. It also makes sense because the compression is not working on event ATM.